woocommerce/woocommerce
# In-App Marketplace: Record the legacy Tracks event for Discover page

Fixes #21486

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Run the `recordMarketplaceView` function whenever the **Extensions > Discover** page loads.

Closes https://github.com/Automattic/woocommerce.com/issues/21486.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch and go to [the in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions)
2. Get a new build (`pnpm install && pnpm build`)
3. Open dev tools console and local storage.setItem( 'debug', 'wc-admin:*' );
4. Reload the page while on the **Discover** tab and confirm you get two Tracks events logged in the console: `wcadmin_marketplace_discover_viewed` and `wcadmin_marketplace_view`
5. Check other tabs: **Browse**, **Themes**, **Business Services** and **My Subscriptions**. Confirm they also log two Tracks events.
6. Please test around this issue



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message


</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Fix missing Tracks event from the Extensions > Discover page

</details>
